File: core/src/com/sheepsgohome/dialogs/NewBadgeDialog.java
package com.sheepsgohome.dialogs;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.ui.Dialog;
import com.badlogic.gdx.scenes.scene2d.ui.Image;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import static com.sheepsgohome.GameData.Loc;
/**
 * Created by LittleLight on 13.1.2015.
 */
public class NewBadgeDialog extends Dialog {

    private final float BUTTON_WIDTH = 50;
    private final float WINDOW_SIZE_MULTIPLICATOR = 1.70f;
    private final float BADGE_SIZE = 100f;
    private TextButton buttonOk;
    private Label title;
    private Label name;

    public NewBadgeDialog(String badgeName, Texture badgeTexture, Skin skin, String windowStyleName) {
        super("", skin, windowStyleName);
        title = new Label(Loc.get("new.badge.earned"), skin);
        name = new Label(badgeName, skin);
        buttonOk = new TextButton(Loc.get("ok"), skin);

        title.setFontScale(0.45f);
        getContentTable().add(title).row();
        getContentTable().add(new Image(badgeTexture)).size(BADGE_SIZE, BADGE_SIZE).row();
        getContentTable().add(name).row();

        buttonOk.getStyle().font.setScale(0.5f);
        buttonOk.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                hide();
            }
        });

        getContentTable().add(buttonOk).size(BUTTON_WIDTH, BUTTON_WIDTH / 2).row();

        initialize();

//        getContentTable().setDebug(true);
//        getButtonTable().setDebug(true);

    }

    private void initialize() {
//        getButtonTable().padBottom(20);
        setModal(true);
        setMovable(false);
        setResizable(false);
    }

    @Override
    protected void result(Object object) {
        this.hide();
    }


    @Override
    public float getPrefWidth() {
        // force dialog width
        return 90 * WINDOW_SIZE_MULTIPLICATOR;
    }

    @Override
    public float getPrefHeight() {
        // force dialog height
        return 140 * WINDOW_SIZE_MULTIPLICATOR;
    }

}

File: core/src/com/sheepsgohome/dialogs/NewBadgeDialog.kt
package com.sheepsgohome.dialogs

import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.ui.TextButton
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.sheepsgohome.badges.Badge
import com.sheepsgohome.shared.GameData.loc
import com.sheepsgohome.shared.GameSkins.skin

class NewBadgeDialog(badgeInstance: Badge, isNew: Boolean = true) : AbstractFixedSizeDialog() {

    private val badge = badgeInstance.clone()

    private val BUTTON_WIDTH = 50f
    private val BADGE_SIZE = 100f

    private val buttonOk: TextButton = TextButton(loc.get("ok"), skin).apply {
        style.font.setScale(0.5f)
    }

    private val titleLabel: Label = Label(loc.get("new.badge.earned"), skin).apply {
        setFontScale(0.45f)
    }

    private val nameLabel: Label = Label(badge.badgeName, skin)

    init {
        fixedHeight = 140f

        if (isNew) {
            contentTable.add(titleLabel).row()
        }

        contentTable.add(badge.image)
                .size(BADGE_SIZE, BADGE_SIZE)
                .row()

        contentTable.add(nameLabel).row()

        buttonOk.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent?, x: Float, y: Float) {
                hide()
            }
        })

        contentTable.add(buttonOk)
                .size(BUTTON_WIDTH, BUTTON_WIDTH / 2)
                .row()
    }

}
